src/o365/spo/commands/listitem/listitem-record-undeclare.ts

```diff
@@ -0,0 +1,305 @@
+        return 'Required parameter webUrl missing';
+      }
+      if (!args.options.id) {
+        return 'Required parameter id missing';
+      }
+
+      const id: number = parseInt(args.options.id);
+      if (isNaN(id)) {
+        return `${args.options.id} is not a valid list item ID`;
+      }
+      const isValidSharePointUrl: boolean | string = SpoCommand.isValidSharePointUrl(args.options.webUrl);
+      if (isValidSharePointUrl !== true) {
+        return isValidSharePointUrl;
+      }
+
+      if (!args.options.listId && !args.options.listTitle) {
+        return `Specify listId or listTitle`;
+      }
+
+      if (args.options.listId && args.options.listTitle) {
+        return `Specify listId or listTitle but not both`;
+      }
+
+      if (args.options.listId &&
+        !Utils.isValidGuid(args.options.listId)) {
+        return `${args.options.listId} in option listId is not a valid GUID`;
+      }
+
+      return true;
+    };
+  }
+
+  public commandHelp(args: {}, log: (help: string) => void): void {
+    const chalk = vorpal.chalk;
+    log(vorpal.find(this.name).helpInformation());
+    log(
+      `  ${chalk.yellow('Important:')} before using this command, log in to a SharePoint Online site,
+    using the ${chalk.blue(commands.LOGIN)} command.
+  
+  Remarks:
+  
+    To undeclare an item as a record in a list, you have to first log in to SharePoint using
+    the ${chalk.blue(commands.LOGIN)} command,
+    eg. ${chalk.grey(`${config.delimiter} ${commands.LOGIN} https://contoso.sharepoint.com`)}.
+        
+  Examples:
+  Undeclare the list item as a record with ID ${chalk.grey(1)} from list with ID
+  ${chalk.grey('0cd891ef-afce-4e55-b836-fce03286cccf')} located in site
+  ${chalk.grey('https://contoso.sharepoint.com/sites/project-x')} 
+    ${chalk.grey(config.delimiter)} ${commands.LISTITEM_RECORD_UNDECLARE} --webUrl https://contoso.sharepoint.com/sites/project-x --listId 0cd891ef-afce-4e55-b836-fce03286cccf --id 1
+
+    Undeclare the list item as a record with ID ${chalk.grey(1)} from list with title
+  ${chalk.grey('List 1')} located in site
+  ${chalk.grey('https://contoso.sharepoint.com/sites/project-x')} 
+    ${chalk.grey(config.delimiter)} ${commands.LISTITEM_RECORD_UNDECLARE} --webUrl https://contoso.sharepoint.com/sites/project-x --listTitle 'List 1' --id 1
+     `);
+  }
+
+  /**
+   * Requests web object identity for the current web.
+   * This request has to be send before we can construct the property bag request.
+   * The response data looks like:
+   * _ObjectIdentity_=<GUID>|<GUID>:site:<GUID>:web:<GUID>
+   * _ObjectType_=SP.Web
+   * ServerRelativeUrl=/sites/contoso
+   * The ObjectIdentity is needed to create another request to retrieve the property bag or set property.
+   * @param webUrl web url
+   * @param cmd command cmd
+   */
+  protected requestObjectIdentity(webUrl: string, cmd: CommandInstance, formDigestValue: string, siteAccessToken: string): Promise<string> {
```

Comment:
There is no need to copy this method. You can reference it directly from https://github.com/pnp/office365-cli/blob/8762b1df83812151ff527ee1df0070b4e1ef5d09/src/o365/spo/common/ClientSvc.ts#L32